### DISCUSSION TITLE
STM32G070RBT6 can not build with zephyr 3.1.99

### DISCUSSION THREAD

**Discussion by xkisahero** [zephyrproject-rtos/zephyr#49151] [Q&A] [answered]
**Describe the bug**
I have update zephyr kernel and zephyr modules yesterday (2022.08.17). Then I found I can't compile projects that can be compiled before.
I have modified the .c file from #include <zephyr.h> to #include <zephyr/zephyr.h>

I have found two questions.
the first is 
[41/177] Building C object CMakeFiles/app.dir/src/wdt_feed_task.c.obj
FAILED: CMakeFiles/app.dir/src/wdt_feed_task.c.obj
E:\my_code\RTOS\zephyr\zephyr-sdk-0.15.0-rc2\arm-zephyr-eabi\bin\arm-zephyr-eabi-gcc.exe -DHSE_VALUE=8000000 -DKERNEL -DSTM32G070xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/include -IE:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0 -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/drivers -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/common -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/soc -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/drivers/include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/drivers/include/Legacy -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/debug/segger/SEGGER -IE:/my_code/RTOS/zephyr/zephyrproject/modules/debug/segger/Config -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/modules/segger/. -isystem E:/my_code/RTOS/zephyr/zephyrproject/zephyr/lib/libc/minimal/include -isystem e:/my_code/rtos/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem e:/my_code/rtos/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=E:/my_code/RTOS/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/arm-zephyr-eabi -imacros E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=E:/my_code/proj_fw/C_knob_V1_0/code=CMAKE_SOURCE_DIR -fmacro-prefix-map=E:/my_code/RTOS/zephyr/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=E:/my_code/RTOS/zephyr/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT CMakeFiles/app.dir/src/wdt_feed_task.c.obj -MF CMakeFiles\app.dir\src\wdt_feed_task.c.obj.d -o CMakeFiles/app.dir/src/wdt_feed_task.c.obj -c E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c
In file included from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/arch/arm/aarch32/arch.h:20,
                 from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/arch/cpu.h:19,
                 from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/kernel_includes.h:33,
                 from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/zephyr.h:18,
                 from E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:1:
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c: In function 'wdt_init':
E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/devicetree_generated.h:6016:46: error: 'DT_N_S_soc_S_watchdog_40002c00_P_label' undeclared (first use in this function); did you mean 'DT_N_S_soc_S_watchdog_40002c00_P_reg'?
 6016 | #define DT_N_INST_0_st_stm32_window_watchdog DT_N_S_soc_S_watchdog_40002c00
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:3091:24: note: in definition of macro 'DT_CAT'
 3091 | #define DT_CAT(a1, a2) a1 ## a2
      |                        ^~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:714:27: note: in expansion of macro 'DT_PROP'
  714 | #define DT_LABEL(node_id) DT_PROP(node_id, label)
      |                           ^~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:30:22: note: in expansion of macro 'DT_LABEL'
   30 | #define WDT_DEV_NAME DT_LABEL(WDT_NODE)
      |                      ^~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_st_stm32_window_watchdog'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:304:31: note: in expansion of macro 'UTIL_CAT'
  304 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:23:18: note: in expansion of macro 'DT_INST'
   23 | #define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
      |                  ^~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:30:31: note: in expansion of macro 'WDT_NODE'
   30 | #define WDT_DEV_NAME DT_LABEL(WDT_NODE)
      |                               ^~~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:69:34: note: in expansion of macro 'WDT_DEV_NAME'
   69 |         wdt = device_get_binding(WDT_DEV_NAME);
      |                                  ^~~~~~~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/devicetree_generated.h:6016:46: note: each undeclared identifier is reported only once for each function it appears in
 6016 | #define DT_N_INST_0_st_stm32_window_watchdog DT_N_S_soc_S_watchdog_40002c00
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:3091:24: note: in definition of macro 'DT_CAT'
 3091 | #define DT_CAT(a1, a2) a1 ## a2
      |                        ^~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:714:27: note: in expansion of macro 'DT_PROP'
  714 | #define DT_LABEL(node_id) DT_PROP(node_id, label)
      |                           ^~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:30:22: note: in expansion of macro 'DT_LABEL'
   30 | #define WDT_DEV_NAME DT_LABEL(WDT_NODE)
      |                      ^~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_st_stm32_window_watchdog'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/devicetree.h:304:31: note: in expansion of macro 'UTIL_CAT'
  304 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:23:18: note: in expansion of macro 'DT_INST'
   23 | #define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
      |                  ^~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:30:31: note: in expansion of macro 'WDT_NODE'
   30 | #define WDT_DEV_NAME DT_LABEL(WDT_NODE)
      |                               ^~~~~~~~
E:/my_code/proj_fw/C_knob_V1_0/code/src/wdt_feed_task.c:69:34: note: in expansion of macro 'WDT_DEV_NAME'
   69 |         wdt = device_get_binding(WDT_DEV_NAME);
      |                                  ^~~~~~~~~~~~
[48/177] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoll.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'E:\my_code\proj_fw\C_knob_V1_0\build'

and the second is 
[104/174] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32g0/soc.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32g0/soc.c.obj
E:\my_code\RTOS\zephyr\zephyr-sdk-0.15.0-rc2\arm-zephyr-eabi\bin\arm-zephyr-eabi-gcc.exe -DHSE_VALUE=8000000 -DKERNEL -DSTM32G070xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/kernel/include -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/include -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/include -IE:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0 -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/drivers -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/common -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/soc -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/drivers/include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/stm32g0xx/drivers/include/Legacy -IE:/my_code/RTOS/zephyr/zephyrproject/modules/hal/stm32/stm32cube/common_ll/include -IE:/my_code/RTOS/zephyr/zephyrproject/modules/debug/segger/SEGGER -IE:/my_code/RTOS/zephyr/zephyrproject/modules/debug/segger/Config -IE:/my_code/RTOS/zephyr/zephyrproject/zephyr/modules/segger/. -isystem E:/my_code/RTOS/zephyr/zephyrproject/zephyr/lib/libc/minimal/include -isystem e:/my_code/rtos/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem e:/my_code/rtos/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=E:/my_code/RTOS/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/arm-zephyr-eabi -imacros E:/my_code/RTOS/zephyr/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=E:/my_code/proj_fw/C_knob_V1_0/code=CMAKE_SOURCE_DIR -fmacro-prefix-map=E:/my_code/RTOS/zephyr/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=E:/my_code/RTOS/zephyr/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32g0/soc.c.obj -MF zephyr\CMakeFiles\zephyr.dir\soc\arm\st_stm32\stm32g0\soc.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32g0/soc.c.obj -c E:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0/soc.c
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0/soc.c: In function 'stm32g0_disable_dead_battery':
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0/soc.c:56:37: error: 'UCPD1_BASE' undeclared (first use in this function); did you mean 'SPI1_BASE'?
   56 |                 if (addr_inst[n] == UCPD1_BASE) {
      |                                     ^~~~~~~~~~
      |                                     SPI1_BASE
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0/soc.c:56:37: note: each undeclared identifier is reported only once for each function it appears in
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/soc/arm/st_stm32/stm32g0/soc.c:61:37: error: 'UCPD2_BASE' undeclared (first use in this function); did you mean 'UID_BASE'?
   61 |                 if (addr_inst[n] == UCPD2_BASE) {
      |                                     ^~~~~~~~~~
      |                                     UID_BASE
[111/174] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/rtt_console.c.obj
ninja: build stopped: subcommand failed.

My PC operating system is window 10.

I think the reason of second question is
zephyrproject\modules\hal\stm32\stm32cube\stm32g0xx\soc\stm32g070xx.h define the macro
#define SYSCFG_CFGR1_UCPD1_STROBE             SYSCFG_CFGR1_UCPD1_STROBE_Msk            /*!< Strobe signal bit for UCPD1 */
#define SYSCFG_CFGR1_UCPD2_STROBE             SYSCFG_CFGR1_UCPD2_STROBE_Msk            /*!< Strobe signal bit for UCPD2 */
but not define UCPD1_BASE
zephyrproject\zephyr\soc\arm\st_stm32\stm32g0\soc.c have following codes:
#if defined(SYSCFG_CFGR1_UCPD1_STROBE)
		if (addr_inst[n] == UCPD1_BASE) {
			strobe &= ~LL_SYSCFG_UCPD1_STROBE;
		}
#endif /* SYSCFG_CFGR1_UCPD1_STROBE */
so bug appears
when I comment the macro in zephyrproject\modules\hal\stm32\stm32cube\stm32g0xx\soc\stm32g070xx.h
// #define SYSCFG_CFGR1_UCPD1_STROBE_Pos         (9U)
// #define SYSCFG_CFGR1_UCPD1_STROBE_Msk         (0x1UL << SYSCFG_CFGR1_UCPD1_STROBE_Pos) /*!< 0x00000200 */
// #define SYSCFG_CFGR1_UCPD1_STROBE             SYSCFG_CFGR1_UCPD1_STROBE_Msk            /*!< Strobe signal bit for UCPD1 */
// #define SYSCFG_CFGR1_UCPD2_STROBE_Pos         (10U)
// #define SYSCFG_CFGR1_UCPD2_STROBE_Msk         (0x1UL << SYSCFG_CFGR1_UCPD2_STROBE_Pos) /*!< 0x00000400 */
// #define SYSCFG_CFGR1_UCPD2_STROBE             SYSCFG_CFGR1_UCPD2_STROBE_Msk            /*!< Strobe signal bit for UCPD2 */
I can build success

**To Reproduce**
I have write a script to build my proess
@echo off
title Zephyr - build project file and devicetree
:: in this file you need modify board name
set board_name=c_knob_v1_0
:: value calaulate
set work_dir=%cd%
set build_dir=\build
set build_dir=%work_dir%%build_dir%
set zephyr_dir=E:\my_code\RTOS\zephyr\zephyrproject
set prj_dir=\code
set prj_dir=%work_dir%%prj_dir%
:: change to zephyrproject folder
cd %zephyr_dir%
:: build
west build --build-dir %build_dir% %prj_dir% --board %board_name%
pause

when I remove the WWDT function and comment the macro, the success build information is list here
-- west build: making build dir E:\my_code\proj_fw\C_knob_V1_0\build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: E:/my_code/proj_fw/C_knob_V1_0/code
-- Found Python3: C:/Python310/python.exe (found suitable exact version "3.10.4") found components: Interpreter
-- Cache files will be written to: E:/my_code/RTOS/zephyr/zephyrproject/zephyr/.cache
-- Zephyr version: 3.1.99 (E:/my_code/RTOS/zephyr/zephyrproject/zephyr)
-- Found west (found suitable version "0.13.0", minimum required is "0.7.1")
-- Board: c_knob_v1_0
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.15.0 (E:/my_code/RTOS/zephyr/zephyr-sdk-0.15.0-rc2)
-- Found dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.5.0", minimum required is "1.4.6")
-- Found toolchain: zephyr 0.15.0 (E:/my_code/RTOS/zephyr/zephyr-sdk-0.15.0-rc2)
-- Found BOARD.dts: E:/my_code/RTOS/zephyr/zephyrproject/zephyr/boards/arm/c_knob_v1_0/c_knob_v1_0.dts
-- Generated zephyr.dts: E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/dts.cmake

warning: HW_STACK_PROTECTION (defined at arch/Kconfig:248) was assigned the value 'y' but got the
value 'n'. Check these unsatisfied dependencies: ARCH_HAS_STACK_PROTECTION (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_HW_STACK_PROTECTION and/or look up
HW_STACK_PROTECTION in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.

warning: WDT_DISABLE_AT_BOOT (defined at drivers/watchdog/Kconfig:14) was assigned the value 'y' but
got the value 'n'. Check these unsatisfied dependencies: WATCHDOG (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_WDT_DISABLE_AT_BOOT and/or look up
WDT_DISABLE_AT_BOOT in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.

Parsing E:/my_code/RTOS/zephyr/zephyrproject/zephyr/Kconfig
Loaded configuration 'E:/my_code/RTOS/zephyr/zephyrproject/zephyr/boards/arm/c_knob_v1_0/c_knob_v1_0_defconfig'
Merged configuration 'E:/my_code/proj_fw/C_knob_V1_0/code/prj.conf'
Configuration saved to 'E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/.config'
Kconfig header saved to 'E:/my_code/proj_fw/C_knob_V1_0/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: E:/my_code/RTOS/zephyr/zephyr-sdk-0.15.0-rc2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc.exe
-- Configuring done
-- Generating done
-- Build files have been written to: E:/my_code/proj_fw/C_knob_V1_0/build
-- west build: building application
[1/174] Generating include/generated/version.h
-- Zephyr version: 3.1.99 (E:/my_code/RTOS/zephyr/zephyrproject/zephyr), build: zephyr-v3.1.0-3609-gf2c49619d269
[33/174] Building C object zephyr/arch/arch/arm/core/aarch...Files/arch__arm__core__aarch32__mpu.dir/arm_core_mpu.c.obj
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c: In function 'z_arm_configure_dynamic_mpu_regions':
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: warning: 'arm_core_mpu_configure_dynamic_mpu_regions' reading 12 bytes from a region of size 0 [-Wstringop-overread]
  311 |         arm_core_mpu_configure_dynamic_mpu_regions(dynamic_regions,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  312 |                                                    region_num);
      |                                                    ~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: note: referencing argument 1 of type 'const struct z_arm_mpu_partition[0]'
In file included from E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:12:
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu_dev.h:187:6: note: in a call to function 'arm_core_mpu_configure_dynamic_mpu_regions'
  187 | void arm_core_mpu_configure_dynamic_mpu_regions(
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: warning: 'arm_core_mpu_configure_dynamic_mpu_regions' reading 12 bytes from a region of size 0 [-Wstringop-overread]
  311 |         arm_core_mpu_configure_dynamic_mpu_regions(dynamic_regions,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  312 |                                                    region_num);
      |                                                    ~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: note: referencing argument 1 of type 'const struct z_arm_mpu_partition[0]'
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu_dev.h:187:6: note: in a call to function 'arm_core_mpu_configure_dynamic_mpu_regions'
  187 | void arm_core_mpu_configure_dynamic_mpu_regions(
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: warning: 'arm_core_mpu_configure_dynamic_mpu_regions' reading 12 bytes from a region of size 0 [-Wstringop-overread]
  311 |         arm_core_mpu_configure_dynamic_mpu_regions(dynamic_regions,
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  312 |                                                    region_num);
      |                                                    ~~~~~~~~~~~
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu.c:311:9: note: referencing argument 1 of type 'const struct z_arm_mpu_partition[0]'
E:/my_code/RTOS/zephyr/zephyrproject/zephyr/arch/arm/core/aarch32/mpu/arm_core_mpu_dev.h:187:6: note: in a call to function 'arm_core_mpu_configure_dynamic_mpu_regions'
  187 | void arm_core_mpu_configure_dynamic_mpu_regions(
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[164/174] Linking C executable zephyr\zephyr_pre0.elf

[168/174] Linking C executable zephyr\zephyr_pre1.elf

[174/174] Linking C executable zephyr\zephyr.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       23420 B       128 KB     17.87%
            SRAM:        8312 B        36 KB     22.55%
        IDT_LIST:          0 GB         2 KB      0.00%

**Comment by nordicjm** [comment, marked as answer]
This is not a bug in zephyr, labels have been removed, you need to update your code to use nodes or aliases or instances
